webapp/channels/src/components/new_search/search_box_suggestions.tsx

```diff
@@ -0,0 +1,159 @@
+
+import type {ProviderResult} from 'components/suggestion/provider';
+import type {SuggestionProps} from 'components/suggestion/suggestion';
+
+import ErrorBoundary from 'plugins/pluggable/error_boundary';
+
+import type {GlobalState} from 'types/store';
+
+const SuggestionsHeader = styled.div`
+    margin-top: 16px;
+    padding: 8px 24px;
+    color: rgba(var(--center-channel-color-rgb), 0.56);
+    font-size: 12px;
+    line-height: 16px;
+    font-weight: 600;
+    text-transform: uppercase;
+`;
+
+const SuggestionsBody = styled.div`
+    border-bottom: 1px solid rgba(var(--center-channel-color-rgb), 0.08);
+    padding-bottom: 16px;
+`;
+
+type Props = {
+    searchType: string;
+    searchTerms: string;
+    setSearchTerms: (searchTerms: string) => void;
+    selectedOption: number;
+    setSelectedOption: (idx: number) => void;
+    suggestionsHeader: React.ReactNode;
+    providerResults: ProviderResult<unknown>|null;
+    focus: (newPosition: number) => void;
+    onSearch: (searchType: string, searchTerms: string) => void;
+}
+
+const SearchSuggestions = ({searchType, searchTerms, setSearchTerms, suggestionsHeader, providerResults, selectedOption, setSelectedOption, focus, onSearch}: Props) => {
+    const license = useSelector(getLicense);
+    const updateSearchValue = useCallback((value: string, matchedPretext: string) => {
+        const changedValue = value.replace(matchedPretext, '');
+        setSearchTerms(searchTerms + changedValue + ' ');
+        focus(searchTerms.length + changedValue.length + 1);
+    }, [searchTerms, setSearchTerms, focus]);
+
+    const runSearch = useCallback((searchTerms: string) => {
+        onSearch(searchType, searchTerms);
+    }, [onSearch, searchType]);
+
+    let SearchPluginSuggestions = useSelector((state: GlobalState) => state.plugins.components.SearchSuggestions) || [];
+    if (license.IsLicensed !== 'true') {
+        SearchPluginSuggestions = [];
+    }
+
+    const generateLabel = (item: any) => {
+        let label = '';
+        if (item.username) {
+            label = item.username;
+            if ((item.first_name || item.last_name) && item.nickname) {
+                label += ` ${item.first_name} ${item.last_name} ${item.nickname}`;
+            } else if (item.nickname) {
+                label += ` ${item.nickname}`;
+            } else if (item.first_name || item.last_name) {
+                label += ` ${item.first_name} ${item.last_name}`;
+            }
+        } else if (item.type === 'D' || item.type === 'G') {
+            label = item.display_name;
+        } else if (item.type === 'P' || item.type === 'O') {
+            label = item.name;
+        } else if (item.emoji) {
+            label = item.name;
+        }
+
+        if (label) {
+            label = label.toLowerCase();
+        }
+        return label;
+    };
+
+    if ((searchType === '' || searchType === 'messages' || searchType === 'files') && providerResults) {
```

Comment:
Is it expected that we always render the plugin component when there's no provider results, even if `searchType` is one of the app-defined ones?

Reply:
I don't fully understand what you mean here. Can you explain it a bit more?

Reply:
I'm trying to understand when we expect the plugin component to be rendered. I thought that the intended use case of this is that the plugin adds its own search type and the component is rendered when the user switches to that search type, but this code looks like the component is also rendered when the user has a Messages or Files search and no results are returned by their search terms.

I wanted to make sure that you did intend the component to appear in both of those cases

Reply:
it should be clearer now.